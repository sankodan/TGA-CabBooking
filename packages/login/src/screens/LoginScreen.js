import React, { useState } from 'react'
import { TouchableOpacity, StyleSheet, View } from 'react-native'
import { Text } from 'react-native-paper'
import { Background, Logo, Header, Button, TextInput, BackButton, Toast } from "@sankodan/com.cabbooking.components"
import { theme } from '@sankodan/com.cabbooking.core'
import { emailValidator } from '@sankodan/com.cabbooking.core'
import { passwordValidator } from '@sankodan/com.cabbooking.core'
import { loginUser } from '@sankodan/com.cabbooking.core'

export default function LoginScreen({ navigation }) {
  const [email, setEmail] = useState({ value: '', error: '' })
  const [password, setPassword] = useState({ value: '', error: '' })
  const [loading, setLoading] = useState()
  const [error, setError] = useState()

  const onLoginPressed = async () => {
    const emailError = emailValidator(email.value)
    const passwordError = passwordValidator(password.value)
    if (emailError || passwordError) {
      setEmail({ ...email, error: emailError })
      setPassword({ ...password, error: passwordError })
      return
    }
    setLoading(true)
    const response = await loginUser({
      email: email.value,
      password: password.value,
    })
    if (response.error) {
      setError(response.error)
    }
    setLoading(false)
  }

  return (
    <Background>
      <BackButton goBack={navigation.goBack} />
      <Logo />
      <Header>Welcome back.</Header>
      <TextInput
        testID="emailText"
        label="Email"
        returnKeyType="next"
        value={email.value}
        onChangeText={(text) => setEmail({ value: text, error: '' })}
        error={!!email.error}
        errorText={email.error}
        autoCapitalize="none"
        autoCompleteType="email"
        textContentType="emailAddress"
        keyboardType="email-address"
      />
      <TextInput
        testID="passwordText"
        label="Password"
        returnKeyType="done"
        value={password.value}
        onChangeText={(text) => setPassword({ value: text, error: '' })}
        error={!!password.error}
        errorText={password.error}
        secureTextEntry
      />
      <View style={styles.forgotPassword}>
        <TouchableOpacity
          testID="ResetPasswordScreen"
          onPress={() => navigation.navigate('ResetPasswordScreen')}
        >
          <Text style={styles.forgot}>Forgot your password?</Text>
        </TouchableOpacity>
      </View>
      <Button testID="loginButton" loading={loading} mode="contained" onPress={onLoginPressed}>
        Login
      </Button>
      <View style={styles.row}>
        <Text>Donâ€™t have an account? </Text>
        <TouchableOpacity testID="RegisterScreen" onPress={() => navigation.replace('RegisterScreen')}>
          <Text style={styles.link}>Sign up</Text>
        </TouchableOpacity>
      </View>
      <Toast testID="onDismiss" message={error} onDismiss={() => setError('')} />
    </Background>
  )
}

const styles = StyleSheet.create({
  forgotPassword: {
    width: '100%',
    alignItems: 'flex-end',
    marginBottom: 24,
  },
  row: {
    flexDirection: 'row',
    marginTop: 4,
  },
  forgot: {
    fontSize: 13,
    color: theme.colors.secondary,
  },
  link: {
    fontWeight: 'bold',
    color: theme.colors.primary,
  },
})
