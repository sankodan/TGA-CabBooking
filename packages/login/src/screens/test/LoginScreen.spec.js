import React from 'react';
import {fireEvent, render, screen} from '@testing-library/react-native';
import {shallow} from 'enzyme';
import  LoginScreen from '../LoginScreen'

jest.mock('@sankodan/com.cabbooking.core/src/api/auth')

describe('LoginScreen', () => {
  describe('clicking send', () => {
    const messageText = 'test@test.com';
    const password = '1234567890';
    let onLoginPressed;

    const navigate = jest.fn();
    const popToTop = jest.fn();
    const push = jest.fn();
    const navigation = {
      navigate,
      goBack: jest.fn(),
      popToTop,
      push
    };

    const wrapper = shallow(<LoginScreen navigation = {navigation}/>);


    beforeEach(() => {
      onLoginPressed = jest.fn().mockName('onLoginPressed');
      render(<LoginScreen navigation = {navigation}/>);
      fireEvent.changeText(screen.getByTestId('emailText'), messageText);
      fireEvent.changeText(screen.getByTestId('passwordText'), password);
      fireEvent.press(screen.getByTestId('loginButton'));
    });

    it('clears the message field', () => {
      expect(screen.getByTestId('emailText')).toHaveProp('value', messageText);
    });
  });
  

  describe('worng password send', () => {
    const messageText = 'test@test.com';
    const password = 'test12345678';
    let onLoginPressed;

    const navigate = jest.fn();
    const popToTop = jest.fn();
    const push = jest.fn();
    const navigation = {
      navigate,
      goBack: jest.fn(),
      popToTop,
      push,
      replace: jest.fn(),
    };

    const wrapper = shallow(<LoginScreen navigation = {navigation}/>);


    beforeEach(() => {
      onLoginPressed = jest.fn().mockName('onLoginPressed');
      let { toJSON } = render(<LoginScreen navigation = {navigation}/>);
      fireEvent.changeText(screen.getByTestId('emailText'), messageText);
      fireEvent.changeText(screen.getByTestId('passwordText'), password);
      fireEvent.press(screen.getByTestId('loginButton'));
      expect(toJSON()).toMatchSnapshot()
    });

    it('clears the message field', () => {

      expect(screen.getByTestId('emailText')).toHaveProp('value', messageText);
    });
  });
  describe('worng password send', () => {
    const messageText = 'test@test.com';
    const password = '123';
    let onLoginPressed;

    const navigate = jest.fn();
    const popToTop = jest.fn();
    const push = jest.fn();
    const navigation = {
      navigate,
      goBack: jest.fn(),
      popToTop,
      push,
      replace: jest.fn(),
    };

    const wrapper = shallow(<LoginScreen navigation = {navigation}/>);


    beforeEach(() => {
      onLoginPressed = jest.fn().mockName('onLoginPressed');
      render(<LoginScreen navigation = {navigation}/>);
      fireEvent.changeText(screen.getByTestId('emailText'), messageText);
      fireEvent.changeText(screen.getByTestId('passwordText'), password);
      fireEvent.press(screen.getByTestId('loginButton'));
      
    });

    it('clears the message field', () => {
      expect(screen.getByTestId('emailText')).toHaveProp('value', messageText);
    });
  });

  describe('click Reset Password', () => {
    const messageText = 'test@test.com';
    const password = '123';
    let onLoginPressed;

    const navigate = jest.fn();
    const popToTop = jest.fn();
    const push = jest.fn();
    const navigation = {
      navigate,
      goBack: jest.fn(),
      popToTop,
      push,
      replace: jest.fn(),
    };

    const wrapper = shallow(<LoginScreen navigation = {navigation}/>);


    beforeEach(() => {
      onLoginPressed = jest.fn().mockName('onLoginPressed');
      render(<LoginScreen navigation = {navigation}/>);
      fireEvent.changeText(screen.getByTestId('emailText'), messageText);
      fireEvent.changeText(screen.getByTestId('passwordText'), password);
      fireEvent.press(screen.getByTestId('ResetPasswordScreen'));
    });

    it('clears the message field', () => {
      expect(screen.getByTestId('emailText')).toHaveProp('value', messageText);
    });
  });

  describe('click RegisterScreen', () => {
    const messageText = 'test@test.com';
    const password = '123';

    const navigate = jest.fn();
    const popToTop = jest.fn();
    const push = jest.fn();
    const navigation = {
      navigate,
      goBack: jest.fn(),
      popToTop,
      push,
      replace: jest.fn(),
    };

    const wrapper = shallow(<LoginScreen navigation = {navigation}/>);


    beforeEach(() => {
      render(<LoginScreen navigation = {navigation}/>);
      fireEvent.changeText(screen.getByTestId('emailText'), messageText);
      fireEvent.changeText(screen.getByTestId('passwordText'), password);
      fireEvent.press(screen.getByTestId('RegisterScreen'));
    });

    it('clears the message field', () => {
      expect(screen.getByTestId('emailText')).toHaveProp('value', messageText);
    });
  });

  describe('click RegisterScreen', () => {
    const messageText = 'test@test.com';
    const password = '123';

    const navigate = jest.fn();
    const popToTop = jest.fn();
    const push = jest.fn();
    const navigation = {
      navigate,
      goBack: jest.fn(),
      popToTop,
      push,
      replace: jest.fn(),
    };

    const wrapper = shallow(<LoginScreen navigation = {navigation}/>);


    beforeEach(() => {
      render(<LoginScreen navigation = {navigation}/>);
      fireEvent.changeText(screen.getByTestId('emailText'), messageText);
      fireEvent.changeText(screen.getByTestId('passwordText'), password);
      fireEvent.press(screen.getByTestId('onDismiss'));
    });

    it('clears the message field', () => {
      expect(screen.getByTestId('emailText')).toHaveProp('value', messageText);
    });
  });


});
