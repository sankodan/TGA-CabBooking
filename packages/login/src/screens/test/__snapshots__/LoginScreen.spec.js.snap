// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen worng password send clears the message field 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
      "width": "100%",
    }
  }
>
  <Image
    resizeMode="repeat"
    source={
      Object {
        "testUri": "../../../packages/components/assets/background_dot.png",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flex": 1,
          "justifyContent": "center",
          "maxWidth": 340,
          "padding": 20,
          "width": "100%",
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "left": 4,
          "opacity": 1,
          "position": "absolute",
          "top": 60,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../packages/components/assets/arrow_back.png",
          }
        }
        style={
          Object {
            "height": 24,
            "width": 24,
          }
        }
      />
    </View>
    <Image
      source={
        Object {
          "testUri": "../../../packages/components/assets/logo.png",
        }
      }
      style={
        Object {
          "height": 110,
          "marginBottom": 8,
          "width": 110,
        }
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Object {
            "color": "#121330",
            "fontSize": 21,
            "fontWeight": "bold",
            "paddingVertical": 12,
          },
        ]
      }
    >
      Welcome back.
    </Text>
    <View
      style={
        Object {
          "marginVertical": 12,
          "width": "100%",
        }
      }
    >
      <View
        style={Object {}}
      >
        <View
          pointerEvents="none"
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 6,
              },
              Object {
                "backgroundColor": "#ffffff",
                "borderColor": "rgba(0, 0, 0, 0.54)",
                "borderRadius": 4,
                "borderWidth": 1,
              },
            ]
          }
          testID="text-input-outline"
        />
        <View>
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 0,
                },
                Object {
                  "minHeight": 64,
                  "paddingTop": 8,
                },
              ]
            }
          >
            <View
              collapsable={false}
              pointerEvents="none"
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                  "zIndex": 3,
                }
              }
            >
              <View
                collapsable={false}
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "#ffffff",
                    "bottom": 4,
                    "left": 10,
                    "maxHeight": 2,
                    "opacity": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 6,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                    ],
                    "width": 12,
                  }
                }
              />
              <Text
                collapsable={false}
                maxFontSizeMultiplier={1.5}
                numberOfLines={1}
                style={
                  Object {
                    "backgroundColor": "#ffffff",
                    "color": "transparent",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "left": 18,
                    "maxWidth": -28,
                    "opacity": 0,
                    "paddingHorizontal": 0,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 37,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                      Object {
                        "scale": 1,
                      },
                      Object {
                        "scaleY": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
              >
                Email
              </Text>
              <Text
                collapsable={false}
                maxFontSizeMultiplier={1.5}
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "#6200ee",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "left": 0,
                    "opacity": 0,
                    "paddingHorizontal": 14,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 36,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                      Object {
                        "scale": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
              >
                Email
              </Text>
              <Text
                collapsable={false}
                maxFontSizeMultiplier={1.5}
                numberOfLines={1}
                style={
                  Object {
                    "color": "rgba(0, 0, 0, 0.54)",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "left": 0,
                    "opacity": 0,
                    "paddingHorizontal": 14,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 36,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                      Object {
                        "scale": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
              >
                Email
              </Text>
            </View>
            <TextInput
              autoCapitalize="none"
              autoCompleteType="email"
              editable={true}
              keyboardType="email-address"
              maxFontSizeMultiplier={1.5}
              multiline={false}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder=""
              placeholderTextColor="rgba(0, 0, 0, 0.54)"
              returnKeyType="next"
              selectionColor="#121330"
              style={
                Array [
                  Object {
                    "flexGrow": 1,
                    "margin": 0,
                    "paddingHorizontal": 14,
                    "zIndex": 1,
                  },
                  Object {
                    "height": 56,
                  },
                  Object {
                    "paddingBottom": 0,
                    "paddingTop": 0,
                  },
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontVariant": undefined,
                    "fontWeight": undefined,
                    "textAlign": "left",
                    "textAlignVertical": "center",
                  },
                  false,
                  Array [
                    Object {},
                  ],
                ]
              }
              testID="emailText"
              textContentType="emailAddress"
              underlineColorAndroid="transparent"
              value="test@test.com"
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "marginVertical": 12,
          "width": "100%",
        }
      }
    >
      <View
        style={Object {}}
      >
        <View
          pointerEvents="none"
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 6,
              },
              Object {
                "backgroundColor": "#ffffff",
                "borderColor": "rgba(0, 0, 0, 0.54)",
                "borderRadius": 4,
                "borderWidth": 1,
              },
            ]
          }
          testID="text-input-outline"
        />
        <View>
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 0,
                },
                Object {
                  "minHeight": 64,
                  "paddingTop": 8,
                },
              ]
            }
          >
            <View
              collapsable={false}
              pointerEvents="none"
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                  "zIndex": 3,
                }
              }
            >
              <View
                collapsable={false}
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "#ffffff",
                    "bottom": 4,
                    "left": 10,
                    "maxHeight": 2,
                    "opacity": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 6,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                    ],
                    "width": 12,
                  }
                }
              />
              <Text
                collapsable={false}
                maxFontSizeMultiplier={1.5}
                numberOfLines={1}
                style={
                  Object {
                    "backgroundColor": "#ffffff",
                    "color": "transparent",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "left": 18,
                    "maxWidth": -28,
                    "opacity": 0,
                    "paddingHorizontal": 0,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 37,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                      Object {
                        "scale": 1,
                      },
                      Object {
                        "scaleY": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
              >
                Password
              </Text>
              <Text
                collapsable={false}
                maxFontSizeMultiplier={1.5}
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "#6200ee",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "left": 0,
                    "opacity": 0,
                    "paddingHorizontal": 14,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 36,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                      Object {
                        "scale": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
              >
                Password
              </Text>
              <Text
                collapsable={false}
                maxFontSizeMultiplier={1.5}
                numberOfLines={1}
                style={
                  Object {
                    "color": "rgba(0, 0, 0, 0.54)",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "left": 0,
                    "opacity": 0,
                    "paddingHorizontal": 14,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 36,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                      Object {
                        "scale": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
              >
                Password
              </Text>
            </View>
            <TextInput
              editable={true}
              maxFontSizeMultiplier={1.5}
              multiline={false}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder=""
              placeholderTextColor="rgba(0, 0, 0, 0.54)"
              returnKeyType="done"
              secureTextEntry={true}
              selectionColor="#121330"
              style={
                Array [
                  Object {
                    "flexGrow": 1,
                    "margin": 0,
                    "paddingHorizontal": 14,
                    "zIndex": 1,
                  },
                  Object {
                    "height": 56,
                  },
                  Object {
                    "paddingBottom": 0,
                    "paddingTop": 0,
                  },
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontVariant": undefined,
                    "fontWeight": undefined,
                    "textAlign": "left",
                    "textAlignVertical": "center",
                  },
                  false,
                  Array [
                    Object {},
                  ],
                ]
              }
              testID="passwordText"
              underlineColorAndroid="transparent"
              value="test12345678"
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "marginBottom": 24,
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="ResetPasswordScreen"
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Object {
                "color": "#414757",
                "fontSize": 13,
              },
            ]
          }
        >
          Forgot your password?
        </Text>
      </View>
    </View>
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "#6200ee",
          "borderColor": "transparent",
          "borderRadius": 4,
          "borderStyle": "solid",
          "borderWidth": 0,
          "elevation": 2,
          "marginVertical": 10,
          "minWidth": 64,
          "paddingVertical": 2,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 0.75,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 1.5,
          "width": "100%",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 4,
            },
          ]
        }
        testID="loginButton"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <View
            accessibilityRole="progressbar"
            accessibilityState={
              Object {
                "busy": true,
              }
            }
            accessible={true}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                Object {
                  "marginLeft": 12,
                  "marginRight": -4,
                },
              ]
            }
          >
            <View
              collapsable={false}
              style={
                Object {
                  "height": 15,
                  "opacity": 1,
                  "width": 15,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "alignItems": "center",
                    "bottom": 0,
                    "justifyContent": "center",
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    Object {
                      "height": 15,
                      "transform": Array [
                        Object {
                          "rotate": "45deg",
                        },
                      ],
                      "width": 15,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      Object {
                        "height": 7.5,
                        "overflow": "hidden",
                        "width": 15,
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      style={
                        Object {
                          "height": 15,
                          "transform": Array [
                            Object {
                              "translateY": 0,
                            },
                            Object {
                              "rotate": "-165deg",
                            },
                          ],
                          "width": 15,
                        }
                      }
                    >
                      <View
                        collapsable={false}
                        style={
                          Object {
                            "height": 7.5,
                            "overflow": "hidden",
                            "width": 15,
                          }
                        }
                      >
                        <View
                          collapsable={false}
                          style={
                            Object {
                              "borderColor": "#ffffff",
                              "borderRadius": 7.5,
                              "borderWidth": 1.5,
                              "height": 15,
                              "width": 15,
                            }
                          }
                        />
                      </View>
                    </View>
                  </View>
                </View>
              </View>
              <View
                collapsable={false}
                style={
                  Object {
                    "alignItems": "center",
                    "bottom": 0,
                    "justifyContent": "center",
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    Object {
                      "height": 15,
                      "transform": Array [
                        Object {
                          "rotate": "45deg",
                        },
                      ],
                      "width": 15,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      Object {
                        "height": 7.5,
                        "overflow": "hidden",
                        "top": 7.5,
                        "width": 15,
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      style={
                        Object {
                          "height": 15,
                          "transform": Array [
                            Object {
                              "translateY": -7.5,
                            },
                            Object {
                              "rotate": "345deg",
                            },
                          ],
                          "width": 15,
                        }
                      }
                    >
                      <View
                        collapsable={false}
                        style={
                          Object {
                            "height": 7.5,
                            "overflow": "hidden",
                            "width": 15,
                          }
                        }
                      >
                        <View
                          collapsable={false}
                          style={
                            Object {
                              "borderColor": "#ffffff",
                              "borderRadius": 7.5,
                              "borderWidth": 1.5,
                              "height": 15,
                              "width": 15,
                            }
                          }
                        />
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "letterSpacing": 1,
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  undefined,
                  Object {
                    "textTransform": "uppercase",
                  },
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontSize": 15,
                    "fontWeight": "bold",
                    "lineHeight": 26,
                  },
                ],
              ]
            }
          >
            Login
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginTop": 4,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "textAlign": "left",
            },
            undefined,
          ]
        }
      >
        Don’t have an account? 
      </Text>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="RegisterScreen"
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Object {
                "color": "#121330",
                "fontWeight": "bold",
              },
            ]
          }
        >
          Sign up
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "position": "absolute",
          "top": 180,
          "width": "100%",
        }
      }
      testID="onDismiss"
    />
  </View>
</View>
`;
